From 85b6a09080fc82d5150fc05096de9996827d783d Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Mon, 16 Oct 2023 01:00:23 -0700
Subject: [PATCH 13/13] 
 0019-Fix-Defaultgateway6-Is-Zero-When-Setting-More-Than-One_IPv6.patch

---
 ...Zero-When-Setting-More-Than-One_IPv6.patch | 31 +++++++++++++++++++
 src/ethernet_interface.cpp                    |  6 ++--
 2 files changed, 34 insertions(+), 3 deletions(-)
 create mode 100644 0019-Fix-Defaultgateway6-Is-Zero-When-Setting-More-Than-One_IPv6.patch

diff --git a/0019-Fix-Defaultgateway6-Is-Zero-When-Setting-More-Than-One_IPv6.patch b/0019-Fix-Defaultgateway6-Is-Zero-When-Setting-More-Than-One_IPv6.patch
new file mode 100644
index 0000000..e2a0b18
--- /dev/null
+++ b/0019-Fix-Defaultgateway6-Is-Zero-When-Setting-More-Than-One_IPv6.patch
@@ -0,0 +1,31 @@
+From 7044480ec7ded1835004d8fe1cd72a401fc8d548 Mon Sep 17 00:00:00 2001
+From: Ethan Wei <ethanwei@ami.com>
+Date: Tue, 11 Jul 2023 13:55:10 +0800
+Subject: [PATCH] Not set gateway6 and retain current value if incoming
+ gateway6 is empty 	modified:   src/ethernet_interface.cpp
+
+---
+ src/ethernet_interface.cpp | 6 +++---
+ 1 file changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
+index 6689f05..2c7a43b 100644
+--- a/src/ethernet_interface.cpp
++++ b/src/ethernet_interface.cpp
+@@ -273,10 +273,10 @@ ObjectPath EthernetInterface::ip(IP::Protocol protType, std::string ipaddress,
+                 ip_address::isValidIPv6Addr((in6_addr*)(&addr.value()), ip_address::Type::IP6_ADDRESS);
+                 if(!ipgateway.empty())
+                 {
+-                        gateway.emplace(stdplus::fromStr<stdplus::In6Addr>(ipgateway));
+-                        ip_address::isValidIPv6Addr((in6_addr*)(&gateway.value()), ip_address::Type::IP6_ADDRESS);
++                    gateway.emplace(stdplus::fromStr<stdplus::In6Addr>(ipgateway));
++                    ip_address::isValidIPv6Addr((in6_addr*)(&gateway.value()), ip_address::Type::IP6_ADDRESS);
++                    EthernetInterfaceIntf::defaultGateway6(ipgateway);
+                 }
+-                defaultGateway6(ipgateway);
+                 break;
+             default:
+                 throw std::logic_error("Exhausted protocols");
+-- 
+2.25.1
+
diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index fa7829e..6ef4753 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -298,10 +298,10 @@ ObjectPath EthernetInterface::ip(IP::Protocol protType, std::string ipaddress,
                 ip_address::isValidIPv6Addr((in6_addr*)(&addr.value()), ip_address::Type::IP6_ADDRESS);
                 if(!ipgateway.empty())
                 {
-                        gateway.emplace(stdplus::fromStr<stdplus::In6Addr>(ipgateway));
-                        ip_address::isValidIPv6Addr((in6_addr*)(&gateway.value()), ip_address::Type::IP6_ADDRESS);
+                    gateway.emplace(stdplus::fromStr<stdplus::In6Addr>(ipgateway));
+                    ip_address::isValidIPv6Addr((in6_addr*)(&gateway.value()), ip_address::Type::IP6_ADDRESS);
+                    EthernetInterfaceIntf::defaultGateway6(ipgateway);
                 }
-                defaultGateway6(ipgateway);
                 break;
             default:
                 throw std::logic_error("Exhausted protocols");
-- 
2.25.1

